share/pot/start.sh

```diff
@@ -153,8 +153,18 @@ _js_vnet()
 	## if norcscript - write a ad-hoc one
 	if [ "$(_get_conf_var "$_pname" "pot.attr.no-rc-script")" = "YES" ]; then
 		touch "${POT_FS_ROOT}/jails/$_pname/m/tmp/tinirc"
-		echo "ifconfig ${_epairb} inet $_ip netmask $POT_NETMASK" >> "${POT_FS_ROOT}/jails/$_pname/m/tmp/tinirc"
-		echo "route add default $POT_GATEWAY" >> "${POT_FS_ROOT}/jails/$_pname/m/tmp/tinirc"
+		cat >>"${POT_FS_ROOT}/jails/$_pname/m/tmp/tinirc" <<-EOT
+		if ! ifconfig ${_epairb} >/dev/null 2>&1; then
+		    sleep 1
+		    if ! ifconfig ${_epairb} >/dev/null 2>&1; then
+		        >&2 echo "Interface ${_epairb} does not exist"
+		        exit 1
+		    fi
+		fi
+		ifconfig ${_epairb} inet $_ip netmask $POT_NETMASK
+		route add default $POT_GATEWAY
+		service ip6addrctl start >/dev/null 2>&1
```

Comment:
While I think it's a great addition to have address selection policy configured, I have doubts about using an rc script here.

`tinirc` was initially designed to run in a jail almost empty, without the rc scripts. Of course, we can change it and relax this condition, if convenient, and make use of those well crafted scripts
However, we would need to have a `/etc/rc.conf` to configure the services. At the moment, the service is not even enabled ,and we don't generate a `rc.conf`

Reply:
Additionally, the address selection policy make sense only when the `dual` stack is selected.
I know that the `rc` script is smart enough to skip everything if no `inet6` is detected, but it seems we could have issue in case of ipv6 only stack, because we don't have other `rc` variables defined.

Reply:
@pizzamig the “service” (it’s actually just calling a command) is enabled by default, so nothing needs to written to rc.conf (this is actually tested).

Like I suggested in the issue, we could alternatively spell out what the script is doing (what is in the issue would work). The reason I called the service command is that it:
1. Allows containers to override it by configuration (in rc.conf) quite easily
2. It is a noop in case the container is really stripped down (no etc dir, even though that’s not a realistic option in most cases).

If we never want to allow 1. (or expect users to do that in pot.cmd), the explicit solution could be used instead.

Reply:
> I guess it could make sense to move the address selection policy in a different PR,
> to cover all the cases: no `tinirc` and all stack selection (ipv4, ipv6, dual) support



> I guess it could make sense to move the address selection policy in a different PR,
> to cover all the cases: no `tinirc` and all stack selection (ipv4, ipv6, dual) support

I think in case of “normal” rc script jails, there is nothing to be done here (as it’s really an in-container config in that case). Unless we want to add a mechanism to alter an rc script based jails stack configuration in an invasive way on pot start. But this would probably a bigger project.

What about this: I change this PR, so that it contains setting ip6addrpolicy without the use of the rc system - this would fix the current issue at hand and as far I can tell, ipv6 is currently not really supported inside of jails anyway, so it shouldn’t break existing setups.

We then can create an issue to discuss what ipv6 based pots could actually look like and what things would need to be done, to properly support them (which sounds more like a long-term project with lower priority).

p.s. https://github.com/freebsd/freebsd-src/blob/3f3e4f3c7472640aab998612a2253da95102345a/libexec/rc/rc.conf#L546

This became an issue since a couple of tools got unified (prominent example is ping). Anything started via rc does the right thing by default though, it’s quite confusing with thin jails, so I’m not a fan of these tools getting unified in the first place.